src/libPersistence/BlockStorage.cpp

```diff
@@ -721,6 +822,8 @@ bool BlockStorage::PutStateDelta(const uint64_t& finalBlockNum,
                                  const bytes& stateDelta) {
   LOG_MARKER();
 
+  unique_lock<shared_timed_mutex> g(m_mutexStateDelta);
+
   if (0 != m_stateDeltaDB->Insert(finalBlockNum, stateDelta)) {
```

Comment:
Maybe let `m_mutexStateDelta` to protect `if (0 != m_stateDeltaDB->Insert(finalBlockNum, stateDelta)) {` only